src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborReader.Map.cs

```diff
@@ -60,5 +71,143 @@ public void ReadEndMap()
                 AdvanceDataItemCounters();
             }
         }
+
+        //
+        // Map decoding conformance
+        //
+
+        private void HandleMapKeyAdded()
+        {
+            Debug.Assert(_currentKeyOffset != null && _curentItemIsKey);
+
+            (int Offset, int Length) currentKeyRange = (_currentKeyOffset.Value, _bytesRead - _currentKeyOffset.Value);
+
+            if (_isConformanceLevelCheckEnabled && CborConformanceLevelHelpers.RequiresSortedKeys(ConformanceLevel))
+            {
+                ValidateSortedKeyEncoding(currentKeyRange);
+            }
+            else if (_isConformanceLevelCheckEnabled && CborConformanceLevelHelpers.RequiresUniqueKeys(ConformanceLevel))
```

Comment:
Since the if and else-if both start with the same check, consider just nesting the ifs.

```C#
if (_isConformanceLevelCheckEnabled)
{
    if (RequiresSorted(...

    else if (RequiresUnique(...
```